spec/models/task_spec.rb

```diff
@@ -284,4 +377,219 @@
+          allow(task).to receive(:contributions).and_return([contribution])
+          allow(contribution).to receive(:destroy).and_return(false)
+          destroy
+        end
+
+        it 'does not destroy the base task' do
+          expect(task).not_to be_destroyed
+        end
+
+        it 'does not destroy the contribution' do
+          expect(contribution).not_to be_destroyed
+        end
+
+        it 'does not destroy the suggestion' do
+          expect(suggestion).not_to be_destroyed
+        end
+      end
+
+      context 'when the base task can not be deleted' do
+        before do
+          allow(task).to receive(:before_committed!).and_raise(ActiveRecord::Rollback)
+        end
+
+        it 'does not destroy the base task' do
+          destroy
+          expect(task).not_to be_destroyed
+        end
+
+        it 'does not destroy the contribution' do
+          destroy
+          expect(contribution).not_to be_destroyed
+        end
+
+        it 'does not destroy the suggestion' do
+          destroy
+          expect(suggestion).not_to be_destroyed
+        end
+
+        it 'calls handle_contributions_on_destroy' do
+          expect(task).to receive(:handle_contributions_on_destroy)
+          destroy
+        end
+      end
+    end
+
+    context 'when a contribution is closed' do
+      let(:status) { :closed }
+
+      it 'destroys the contribution' do
+        expect { destroy }.to change(TaskContribution, :count).by(-1)
+        expect(contribution).to be_destroyed
+      end
+
+      it 'destroys the suggestion' do
+        destroy
+        expect(suggestion).to be_destroyed
+      end
+
+      it 'destroys the base task' do
+        destroy
+        expect(task).to be_destroyed
+      end
+    end
+  end
+
+  describe '#apply_contribution' do
+    subject(:apply_contribution) { task.apply_contribution(contribution) }
+
+    let(:task) { create(:task) }
+    let(:contribution) { create(:task_contribution, base: task) }
+
+    before do
+      contribution.suggestion.title = 'New title'
+      contribution.suggestion.meta_data = 'New meta data'
+      contribution.suggestion.submission_restrictions = 'New submission restrictions'
+      contribution.suggestion.external_resources = 'New external resources'
+      contribution.suggestion.grading_hints = 'New grading hints'
+    end
+
+    context 'when the contribution is pending' do
+      it 'applies the contribution' do
+        expect(apply_contribution).to be(true)
+      end
+
+      it 'closes the contribution' do
+        apply_contribution
+        expect(contribution.status_merged?).to be(true)
+      end
+
+      it 'transfers the metadata' do
+        apply_contribution
+        expect(task.meta_data).to eq('New meta data')
+        expect(task.submission_restrictions).to eq('New submission restrictions')
+        expect(task.external_resources).to eq('New external resources')
+        expect(task.grading_hints).to eq('New grading hints')
```

Comment:
Here is a patch for setting real-world meta data:

<details><summary>Details</summary>
<p>

```diff
Subject: [PATCH] Set ProFormA meta data in contributions
---
Index: spec/models/task_spec.rb
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/spec/models/task_spec.rb b/spec/models/task_spec.rb
--- a/spec/models/task_spec.rb	(revision 0d282a9a941e27556ce306e5d2215cf7bd7cce3f)
+++ b/spec/models/task_spec.rb	(date 1726044898071)
@@ -542,12 +542,17 @@
     let(:task) { create(:task) }
     let(:contribution) { create(:task_contribution, base: task) }
 
+    let(:meta_data) { attributes_for(:task, :with_meta_data)[:meta_data] }
+    let(:submission_restrictions) { attributes_for(:task, :with_submission_restrictions)[:submission_restrictions] }
+    let(:external_resources) { attributes_for(:task, :with_external_resources)[:external_resources] }
+    let(:grading_hints) { attributes_for(:task, :with_grading_hints)[:grading_hints] }
+
     before do
       contribution.suggestion.title = 'New title'
-      contribution.suggestion.meta_data = 'New meta data'
-      contribution.suggestion.submission_restrictions = 'New submission restrictions'
-      contribution.suggestion.external_resources = 'New external resources'
-      contribution.suggestion.grading_hints = 'New grading hints'
+      contribution.suggestion.meta_data = meta_data
+      contribution.suggestion.submission_restrictions = submission_restrictions
+      contribution.suggestion.external_resources = external_resources
+      contribution.suggestion.grading_hints = grading_hints
     end
 
     context 'when the contribution is pending' do
@@ -562,10 +567,10 @@
 
       it 'transfers the metadata' do
         apply_contribution
-        expect(task.meta_data).to eq('New meta data')
-        expect(task.submission_restrictions).to eq('New submission restrictions')
-        expect(task.external_resources).to eq('New external resources')
-        expect(task.grading_hints).to eq('New grading hints')
+        expect(task.meta_data).to eq meta_data
+        expect(task.submission_restrictions).to eq submission_restrictions
+        expect(task.external_resources).to eq external_resources
+        expect(task.grading_hints).to eq grading_hints
       end
 
       it 'transfers the labels' do
Index: spec/controllers/task_contributions_controller_spec.rb
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/spec/controllers/task_contributions_controller_spec.rb b/spec/controllers/task_contributions_controller_spec.rb
--- a/spec/controllers/task_contributions_controller_spec.rb	(revision 0d282a9a941e27556ce306e5d2215cf7bd7cce3f)
+++ b/spec/controllers/task_contributions_controller_spec.rb	(date 1726044748295)
@@ -8,7 +8,7 @@
   let(:contribution_user) { create(:user) }
   let(:user) { contribution_user }
   let(:original_author) { create(:user) }
-  let!(:task) { create(:task, user: original_author, access_level: 'public') }
+  let!(:task) { create(:task, :with_meta_data, :with_submission_restrictions, :with_external_resources, :with_grading_hints, user: original_author, access_level: 'public') }
   let(:contrib_task) { build(:task, user: contribution_user, title: 'Modified title', parent_uuid: task.uuid) }
   let(:contribution) { build(:task_contribution, suggestion: contrib_task, base: task, status: :pending) }
 ```

</p>
</details>

Reply:
I have applied your patch.